core/iwasm/common/gc/gc_common.c

```diff
@@ -53,6 +81,40 @@ wasm_func_type_get_param_count(WASMFuncType *const func_type)
     return 0;
 }
 
+bool
+wasm_ref_type_normalize(wasm_ref_type_t *ref_type)
+{
+    wasm_value_type_t value_type = ref_type->value_type;
+    int32 heap_type = ref_type->heap_type;
+
+    if (value_type < VALUE_TYPE_NULLREF || value_type > VALUE_TYPE_I32) {
+        return false;
+    }
+    if (value_type == VALUE_TYPE_HT_NULLABLE_REF
+        || value_type == VALUE_TYPE_HT_NON_NULLABLE_REF) {
+        if (heap_type < 0) {
+            if (heap_type < HEAP_TYPE_NONE || heap_type > HEAP_TYPE_FUNC) {
```

Comment:
Just using `if (heap_type <= HEAP_TYPE_NONE || heap_type >= HEAP_TYPE_FUNC)`, no need to add `if (heap_type < 0)`, these heap types are already `< 0`